### DISCUSSION TITLE
Flow - Run actions in the client browser

### DISCUSSION THREAD

**Discussion by bernatvadell** [directus/directus#20595] [Feature Requests]
### Summary

Currently, when a workflow completes, a message appears stating that the workflow has completed.

The idea would be to be able to customize the type of report that we want to be made when the flows are "no-async".

These reports could be a:
- Dialog (a personalized message would be displayed)
- Download file (the download of a file would start)

### Motivation

In my case, my custom flow generates a report that is stored as a file. I would like this report to be downloadable directly from the browser. (I currently send it by email to the user who requested the flow).

### Detailed Design

Here the different types of "final" actions could be added. They could be separated from the rest with a Divider.

![image](https://github.com/directus/directus/assets/17298604/20053142-3a36-4fd7-a29f-e9e4b4780a0f)

On the client side, when the execution is completed, we could return the result of the execution of the flow, if it is "async" or does not have a "final" action, the usual thing will be done (the notify). If it is "no-async" and contains a final report action. Depending on the type, one thing or another will be done.
![image](https://github.com/directus/directus/assets/17298604/1b88d3ea-db22-4f92-b647-f6b6fee3fff4)

### Must Have:
- Flow: Resolver dialog custom message
- App: Show dialog custom message
- Flow: Resolver download file
- App: Download file specified and show confirmation message

### Could Have:
- It would be nice to be able to create our own extensions to extend the functionality of the flows' "resolvers".

### Drawbacks

I can't think of any reason why it shouldn't be done:
- It is adaptable to Directus standards
- Development times must be minimal, reusing the current scheme.
- It would increase the possibilities of using the flows by being able to notify the user of a personalized result when the flow is completed.

### Alternatives

Send email

### Adoption Strategy

It does not imply an impact on current users. Since in the event that this type of "resolve" was not specified, it will do the usual thing (notify of the execution of the flow)

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
